doc/releases/changelog-dev.md

```diff
@@ -359,6 +358,19 @@
   RZ**2(1.0, wires=[0])
   ```
 
+* Added `__mul__` and `__matmul__` dunder methods to the `qml.operation.Operator` class so
+  that users can combine operators more naturally.
+  [(#2891)](https://github.com/PennyLaneAI/pennylane/pull/2891)
+
+  ```pycon
+  >>> prod_op = qml.PauliX(0) @ qml.PauliX(1)
```

Comment:
Might want to demonstrate `@` with non-observables, since we could already do `qml.PauliX(0) @ qml.PauliX(1)`.

```suggestion
  >>> prod_op = qml.RX(1, wires=0) @ qml.RY(2, wires=0)
```